File: samcli/local/init/templates/cookiecutter-aws-sam-hello-ruby/{{cookiecutter.project_name}}/hello_world/tests/unit/test_handler.rb
require 'json'
require 'test/unit'
require 'mocha/test_unit'
require_relative '../../app'

class HelloWorldTest < Test::Unit::TestCase

  def test_lambda_handler
    response = lambda_handler(event:"", context:"")
    json_body = JSON.parse(response[:body])

    assert_equal(200, response[:statusCode])
    assert_equal("hello world", json_body["message"])
  end
end

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs6/{{cookiecutter.project_name}}/hello-world/tests/unit/test-handler.js
'use strict';

const app = require('../../app.js');
const chai = require('chai');
const expect = chai.expect;
var event, context;

describe('Tests Handler', function () {
    it('verifies successful response', function (done) {
        app.lambdaHandler(event, context, function (err, result) {
            try {
                expect(result).to.be.an('object');
                expect(result.statusCode).to.equal(200);
                expect(result.body).to.be.an('string');

                let response = JSON.parse(result.body);

                expect(response).to.be.an('object');
                expect(response.message).to.be.equal("hello world");
                // expect(response.location).to.be.an("string");
                done();
            } catch (e) {
                done(e);
            }
        });
    });
});
